docs/user/api/v3.rst

```diff
@@ -186,6 +186,8 @@ Projects list
             }]
         }
 
+    :query string name: return projects with matching name
+    :query string slug: return projects with matching slug
```

Comment:
Does "with matching ..." conveys that it could only partially match? I didn't get at first without reading the code.

Reply:
I copied this from the Version API docs for consistency, but agree both could be clearer. I almost wanted to have fields `name` and `name__icontains`, but just followed what the Version related API pieces did.

Reply:
I don't have strong opinions here, both certainly work. I'll defer to you on this one as you are most familiar with the API construction.

Reply:
I'd follow your approach (using `__` on filters) because I think it's more flexible and will allow better support of features in the future.

I think it's better to do it now without breaking changes than later.

Reply:
Makes sense. Should I keep the `name` filter as an exact match too you think? I will be using the fuzzy match for the dashboard, but exact match on name/slug also seems useful for API users.

I won't touch `Version` filters, which will differ slightly in function (`name` is fuzzy match on `Version`)

Reply:
> Should I keep the name filter as an exact match too you think?

I'd say "No" here so all the endpoint are consistent. `name` will be treated as an alias of `name__icontains` to keep backward compatibility and a standardized way among all the endpoints.

Reply:
Hrm, so I suppose this PR is then accurate then -- that is, leave the `name` filter as `name__icontains`? Or are you describing something else here too?

Perhaps it helps to treat my suggestion as a next iteration and perhaps spend some time on a wider solution. Perhaps django-filters is a solution to weight here?

Reply:
This PR is fine, yes 👍🏼 

> Perhaps it helps to treat my suggestion as a next iteration and perhaps spend some time on a wider solution. Perhaps django-filters is a solution to weight here?

If you are not going to implement the `__<filter>` in this PR, we should open an issue with the plan we talked here --so, we don't deviate from it into an incompatible direction when we need a new filter.

Reply:
Actually, besides the issue, you can add a comment in the code explaining this too 😄

Reply:
Good plan. I opened https://github.com/readthedocs/readthedocs.org/issues/9862 to track this and updated the filter with a comment for later.